sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -244,28 +245,31 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    if (!request.tlsSettings && request.disableKeepAlive) {
+      return isInsecure ? http.globalAgent : https.globalAgent;
+    }
+    const settingsIndex = request.tlsSettings ?? {};
```

Comment:
since the default is an inline object still, won't this generate a new agent each time if no tlsSettings are passed and disableKeepAlive is false? Maybe we should handle keepAlive/noKeepAlive without TLS settings as special? or at least cache them consistently